src/compiler/factory.ts

```diff
@@ -1707,4 +1707,56 @@ namespace ts {
         nodes.hasTrailingComma = hasTrailingComma;
         return nodes;
     }
+
+    export function getLocalNameForExternalImport(node: ImportDeclaration | ExportDeclaration | ImportEqualsDeclaration, sourceFile: SourceFile): Identifier {

```

Comment:
`sourceFile` first; consider renaming or providing JSDoc to explain what this does. At a glance, I find myself asking

1. What makes an import *external*?
2. What exactly is meany by a *local name*?